pkg/injector/profiling.go

```diff
@@ -1,32 +1,43 @@
 package injector
 
 import (
+	"fmt"
 	"net/http"
 	"time"
+
+	"github.com/openservicemesh/osm/pkg/metricsstore"
+)
+
+var (
+	defaultK8sTimeout = time.Duration(30 * time.Second)
 )
 
 // Helper to parse timeout variable from webhook URL
-func readTimeout(req *http.Request) (*time.Duration, error) {
+func readTimeout(req *http.Request) (time.Duration, error) {
 	durationValue, found := req.URL.Query()[webhookMutateTimeoutKey]
 	if !found || len(durationValue) != 1 {
 		log.Error().Msgf("Webhook timeout value not found in request")
-		return nil, errParseWebhookTimeout
+		return defaultK8sTimeout, errParseWebhookTimeout
 	}
 
 	val, err := time.ParseDuration(durationValue[0])
 	if err != nil {
 		log.Error().Msgf("Error parsing timeout value as duration: %v", err)
-		return nil, errParseWebhookTimeout
+		return defaultK8sTimeout, errParseWebhookTimeout
 	}
-	return &val, nil
+	return val, nil
 }
 
 // Time tracking function for webhook processing.
 // Will calculate elapsed time since start and log debug how much time spent executing
-func webhookTimeTrack(start time.Time, timeout time.Duration) {
+func webhookTimeTrack(start time.Time, timeout time.Duration, success *bool) {
 	elapsed := time.Since(start)
 	percentOfTimeout := float64(elapsed.Microseconds()) / float64(timeout.Microseconds())
 
 	log.Debug().Msgf("Mutate Webhook took %v to execute (%.2f of it's timeout, %v)",
 		elapsed, percentOfTimeout, timeout)
+
+	metricsstore.DefaultMetricsStore.InjectorSidecarCount.Inc()
+	metricsstore.DefaultMetricsStore.InjectorRqTime.
+		WithLabelValues(fmt.Sprintf("%t", *success)).Observe(elapsed.Seconds())
```

Comment:
Out of context, this pointer dereference seems dangerous, but I think I understand why `success` is a pointer and why it's not an issue.

Reply:
golang should ensure to keep it around even out of scope as soon as dereference it, till the defer goes out (since it's the only pointer left to it).
I did this consciously expecting for this behavior at least.